File: node-app/user_operations.test.ts
import { CosmosClient, Container, Database } from "@azure/cosmos";
import { createUser, readUser, User } from "./user_operations";
import assert from "assert";

const connectionString = process.env.COSMOSDB_CONNECTION_STRING;
const databaseId = process.env.COSMOSDB_DATABASE_NAME;
const containerId = process.env.COSMOSDB_CONTAINER_NAME;

// dummy comment
if (!connectionString) {
    throw new Error("Please set COSMOSDB_CONNECTION_STRING environment variable.");
}

const client = new CosmosClient(connectionString);
let database: Database;
let container: Container;

async function runTests() {
    const testUser: User = {
        id: "42",
        email: "user42@example.com",
        active: true
    };

    try {
        // Create database and container
        database = (await client.databases.createIfNotExists({ id: databaseId })).database;
        container = (await database.containers.createIfNotExists({ id: containerId, partitionKey: "/id" })).container;

        // Test createUser
        try {
            await createUser(container, testUser);
            console.log("User creation test passed!");
        } catch (error) {
            assert.fail("User creation test failed: " + error);
        }

        // Test readUser
        try {
            const user = await readUser(container, testUser.id);
            assert.equal(testUser.id, user.id)
            assert.equal(testUser.email, user.email)
            assert.equal(testUser.active, user.active)
            console.log("User read test passed!");

        } catch (error) {
            assert.fail("User read test failed: " + error);
        }

    } catch (error) {
        assert.fail("User read test failed: " + error);
    } finally {
        // Delete the database
        if (database) {
            await database.delete();
            console.log("database deleted");
        }
    }
}

runTests();

File: go-app/user/user_operations_test.go
package user

import (
	"context"

	//"demo/user"
	"log"
	"os"
	"testing"

	"github.com/Azure/azure-sdk-for-go/sdk/data/azcosmos"
	"github.com/stretchr/testify/require"
)

var container *azcosmos.ContainerClient

func TestMain(m *testing.M) {

	ctx := context.Background()

	connectionString := os.Getenv("COSMOSDB_CONNECTION_STRING")

    client, err := azcosmos.NewClientFromConnectionString(connectionString, nil)
    if err != nil {
		log.Fatalf("failed to create client: %v", err)
	}

	_, err = client.CreateDatabase(ctx, azcosmos.DatabaseProperties{ID: os.Getenv("COSMOSDB_DATABASE_NAME")}, nil)
    if err != nil {
		log.Fatalf("failed to create database: %v", err)
	}

	log.Println("Database created")


	db, err := client.NewDatabase(os.Getenv("COSMOSDB_DATABASE_NAME"))
	if err != nil {
		log.Fatalf("failed to get database: %v", err)
	}

	_, err = db.CreateContainer(ctx, azcosmos.ContainerProperties{ID: os.Getenv("COSMOSDB_CONTAINER_NAME"), PartitionKeyDefinition: azcosmos.PartitionKeyDefinition{Paths: []string{"/id"}}}, nil)

	if err != nil {
		log.Fatalf("failed to create container: %v", err)
	}

	log.Println("Container created")

	container, err = client.NewContainer(os.Getenv("COSMOSDB_DATABASE_NAME"), os.Getenv("COSMOSDB_CONTAINER_NAME"))
	if err != nil {
		log.Fatalf("failed to get container: %v", err)
	}

	m.Run()

	_, err = db.Delete(ctx, nil)
	if err != nil {
		log.Fatalf("failed to delete database: %v", err)
	}

	log.Println("Database deleted")

}

func TestCreate(t *testing.T) {
    ctx := context.Background()

	var newUser = User{
		ID: "42",
		Email: "user42@test.com",
		Active: true,
	}

	err := Create(ctx, container, newUser)

    require.NoError(t, err)
}

func TestReadItem(t *testing.T) {
    ctx := context.Background()
	
	userID := "43"

	var newUser = User{
		ID: userID,
		Email: "user43@test.com",
		Active: true,
	}

	err := Create(ctx, container, newUser)
    require.NoError(t, err)

    userItem, err := Read(ctx, container, userID)

    require.NoError(t, err)
	require.Equal(t, userID, userItem.ID)
	require.Equal(t, "user43@test.com", userItem.Email)
	require.True(t, true, userItem.Active)
}

// func NewCosmosClient(connectionString string) (*azcosmos.Client, error) {
//     // Create a custom HTTP client with InsecureSkipVerify set to true
//     customTransport := &http.Transport{
//         TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
//     }
//     customHTTPClient := &http.Client{Transport: customTransport}

//     // Create a Cosmos DB client with the custom HTTP client
//     clientOptions := azcosmos.ClientOptions{
// 		ClientOptions: azcore.ClientOptions{
			
// 		},
//     }
//     client, err := azcosmos.NewClientFromConnectionString(connectionString, &clientOptions)
//     if err != nil {
//         return nil, err
//     }
//     return client, nil
// }